src/dynsight/_internal/vision/video_to_frame.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+import cv2
+
+if TYPE_CHECKING:
+    import pathlib
+
+    import numpy as np
+
+
+@dataclass
+class Video:
+    """Load a video file and provides utilities.
+
+    Attributes:
+        video_path:
+            File path to the video.
+    """
+
+    video_path: pathlib.Path
+    frames: list[np.ndarray] = field(default_factory=list)
+
+    def count_frame(self) -> int:
+        """Counts the total number of frames in the video.
+
+        Opens the video file at `video_path` and retrieves the frame count
+        from the video metadata.
+
+        Returns:
+            The number of frames in the video.
+
+        Raises:
+            ValueError:
+                If the video cannot be loaded.
+        """
+        capture = cv2.VideoCapture(str(self.video_path))
+        if not capture.isOpened():
```

Comment:
Do not try to open things with file types that we know cannot be opened? add a check prior? in case trying to open causes a break?